src/sql/parts/disasterRecovery/restore/common/restoreServiceImpl.ts

```diff
@@ -153,9 +155,15 @@ export class RestoreDialogController implements IRestoreDialogController {
 			restoreOption.taskExecutionMode = TaskExecutionMode.executeAndScript;
 		}
 
-		this._restoreService.restore(this._ownerUri, restoreOption);
-		let restoreDialog = this._restoreDialogs[this._currentProvider];
-		restoreDialog.close();
+		this._restoreService.restore(this._ownerUri, restoreOption).then(() => {
+			let connectionProfile = this._connectionService.getConnectionProfile(this._ownerUri);
+			let activeNode = this._objectExplorerService.getObjectExplorerNode(connectionProfile);
+			this._objectExplorerService.refreshTreeNode(activeNode.getSession(), activeNode).then(() => {
```

Comment:
It's possible to have a dashboard open for a connection that doesn't have a corresponding OE node (for example by opening a new connection with "None" as the server group), so we should check if `activeNode` is undefined before using it